src/hooks/useClickOutside.tsx

```diff
@@ -0,0 +1,23 @@
+import react, {useEffect, useRef } from 'react';
+
+export type outClickType = (eve: MouseEvent) => void;
+export function useClickOutside(elRef : any , callback : outClickType) {
```

Comment:
This looks great. Can you think of any other places we might be able to use this? I can see it being useful but I'm not sure where exactly. If I can keep those ideas in mind I think it will help some of my design choices 😄

Reply:
Well this is an idea from the video I shared about custom hooks. 
I that video it is proposed to be used with a menu or with drawer.
I hope is useful It was a pain to implement with typescript but I learned a bunch about refs 🤣 

https://youtu.be/J-g9ZJha8FE?t=483

Reply:
Wow, yeah... that's _extremely_ useful! All of a sudden I can see a ton of different things this is going to be useful for. 🥳